scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/DaemonEnvironment.java

```diff
@@ -48,7 +53,24 @@ public StartTask startTaskFor(ContainerInstance i) {
         .build();
   }
 
-  public boolean matchesTask(Task t) {
+  public StopTask stopTaskFor(final Task task) {
+    return StopTask.builder()
+        .clusterName(environment.getClusterName())
+        .task(task.getTaskDefinitionArn())
+        .reason(
+            String.format(
+                "Stopped by deployment to %s@%s",
+                environment.getEnvironmentName(), environment.getTaskDefinitionArn()))
+        .build();
+  }
+
+  public boolean taskToStop(Task t) {
```

Comment:
How about `isTaskStoppable`?

Reply:
Sure. I was struggling with naming. We should come up with some naming conventions?

Reply:
Yeah. I think in general, the problem is that we are trying to interrogate properties of the instance/tasks. If we were to express these questions naturally would say (roughly)

```java
instances.filter((instance) -> instance.needsNewTask()).map(startTask);
tasks.filter((task) -> task.canBeStopped()).map(stopTask);
```

Since we don't have `needsNewTask` and `canBeStopped` methods on those entities (because we need the `DaemonEnvironment`'s data to make that decision), we have to basically invert the names, and prefix them with what the method receiver would have been:

```java
instance.filter((instance) -> env.instanceNeedsNewTask(instance)).map(startTask)
tasks.filter((task) -> env.taskCanBeStopped(task)).map(stopTask)
```

That gets tricky when e.g. `env.taskCanBeStopped(task)` depends on the instance that it's on, as will be the case for the next deployment method, TerminateAfterReplace.

Reply:
You point out my pain correctly :-D  I have the feeling that in DaemonEnvironment we're inverting the names. I can go with the current naming (of course taking your suggestion) and update them when we finish the TerminateAfterReplace.